spec/functional/knife/rehash_spec.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Author:: Daniel DeLeo (<dan@chef.io>)
+# Copyright:: Copyright 2010-2016, Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+require "chef/knife/rehash"
+require "chef/knife/core/subcommand_loader"
+
+class Chef
+  class Knife
+    class SubcommandLoader
+      def nil_load_commands
+      end
+    end
+  end
+end
+
+describe "knife rehash" do
+  before(:all) do
+    # we don't actually want to use Kernel.load in the class because the paths aren't setup now
+    Chef::Knife::SubcommandLoader.send(:alias_method, :orig_load_commands, :load_commands)
+    Chef::Knife::SubcommandLoader.send(:alias_method, :load_commands, :nil_load_commands)
```

Comment:
Ah, I meant to make a comment about this.  I was trying to avoid mocking out a `Kernel.load` in that method, but your suggestion makes total sense.  Thanks!